code/components/extra-natives-five/src/VehicleExtraNatives.cpp

```diff
@@ -363,4 +367,120 @@ static HookFunction initFunction([]()
 	{
 		*reinterpret_cast<float *>(wheelAddr + WheelXOffsetOffset) = context.GetArgument<float>(2);
 	}));
+
+	static struct : jitasm::Frontend
+	{
+		static bool ShouldSkipRepairFunc(fwEntity* VehPointer)
+		{
+			return g_skipRepairVehicles.find(VehPointer) != g_skipRepairVehicles.end();
+		}
+		virtual void InternalMain() override
+		{
+			//restore rcx after asm call
+			mov(rcx, rbx);
+			//save registers
+			push(rax);
```

Comment:
you're replacing a `call`, so shouldn't have to save any registers at all (other than `rax` you use yourself), as the compiler already assumes these registers will not be saved.

do make sure to keep stack alignment to 16, so remove registers in pairs of two, or change the stack sub/add alignment.